cumulusci/tasks/release_notes/generator.py

```diff
@@ -22,6 +22,12 @@ def __init__(self):
         self.init_parsers()
         self.init_change_notes()
         self.version_id = None
+        self.release_info = False
+        self.trial_info = False
+        self.sandbox_date = datetime.date.today().isoformat()
+        self.production_date = (
+            datetime.date.today() + datetime.timedelta(days=6)
+        ).isoformat()
```

Comment:
I still don't like hardcoding default dates. Using today as the default for sandbox_date is usually okay for our normal release process. But remember that the github_release_notes task can be run again manually in the future to re-generate the release notes if corrections were made on the pull requests. With default dates here, that would change the dates in the release notes!

I think the path we need to follow is:
- Make the task accept options for sandbox_date and production_date, but don't pick any dates by default.
- Make MetaCI pick good default dates when creating a Release object in MetaCI. Then make sure those get passed by MetaCI when running the release_production flow in a way that can be used by this task.